Bulls-and-cows/src/main/java/study/StringCalculator.java

```diff
@@ -0,0 +1,49 @@
+package study;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class StringCalculator {
+
+    private static final int ZERO = 0;
+    private static final int ONE = 1;
+
+    public static int calculate(String input, String delimiter) {
+        return calculate(
+                extract(input.split(delimiter), StringCalculator::isNumber, Integer::parseInt),
+                extract(input.split(delimiter), StringCalculator::isNotNumber, String::valueOf)
+        );
+    }
```

Comment:
```suggestion
    public static int calculate(String input, SplitSrategy strategy) {
        return calculate(
                extract(strategy.split(input), StringCalculator::isNumber, Integer::parseInt),
                extract(strategy.split(input), StringCalculator::isNotNumber, String::valueOf)
        );
    }
```

어제도 말씀드렸지만, delimeter에 대한 강결합이 진행되고 있다고 생각이 들어요  
다른 요소들에서도 전략패턴을 사용했듯이 
SplitStrategy 를 만들어서, 분리하는 작업을 진행해보면 어떨까요?

Reply:
더불어서 Calculate가 문자열을 쪼개는 역할도 애매할 것 같아요
이 부분은 어떻게 분리하면 좋을지 생각해보시면 좋을 것 같아요